k3s.yaml

```diff
@@ -0,0 +1,151 @@
+      - zstd
+      - yq
+      - runc # for buildtime version info
+      - containerd # for buildtime version info
+      - crictl # for buildtime version info
+      - crane
+
+var-transforms:
+  - from: ${{package.version}}
+    match: \_
+    replace: +
+    to: mangled-package-version
+
+# Upstream uses `dapper` to initialize build environments, but since melange
+# already provides a consistent build environment, we adopt upstreams
+# ./scripts as much as possible. Additionally, we try to replace all k3s
+# built third party components with Wolfi replacements.
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/k3s-io/k3s
+      tag: v${{vars.mangled-package-version}}
+      expected-commit: 213d7ad499e166290872f51c63d8eaa2f1fe78b3
+  - uses: patch
+    with:
+      patches: 0001-PATCH-Wolfi-support.patch
+  - runs: |
+      # Patch doesn't include gomod updates
+      go mod tidy
+
+      mkdir -p build/static bin/aux etc
+
+      # The patched version of this just downloads the boot chart manifests.
+      # The buildroot, runc, and containerd are all replaced with wolfi variants
+      # at runtime
+      ./scripts/download
+
+      # The patched version keeps the forked cni-plugins since it conveniently
+      # packages them into a single multicall binary with flannel. However, the
+      # builtin kubectl, containerd, ctr, and crictl are removed in favor of the
+      # wolfi versions.
+      export VERSION_CONTAINERD=$(containerd --version | head -n1 | awk '{print $3}')
+      export VERSION_RUNC=$(runc --version | head -n1 | awk '{print $3}')
+      export VERSION_CRICTL=$(crictl --version | head -n1 | awk '{print $3}')
+      ./scripts/build
+
+      # This builds the multicall binary without the embedded buildroot and tooling.
+      ./scripts/package-cli
+  - runs: |
+      install -Dm755 dist/artifacts/k3s-$(go env GOARCH) "${{targets.destdir}}"/usr/bin/k3s
+
+      # Some distributors require k3s at /bin/k3s, so symlink it here
+      mkdir -p "${{targets.destdir}}"/bin
+      ln -s ../usr/bin/k3s "${{targets.destdir}}"/bin/k3s
+
+  - uses: strip
+
+subpackages:
+  - name: k3s-data
+    description: "k3s data directory, pre-extracted"
+    dependencies:
+      runtime:
+        - k3s
+    pipeline:
+      - runs: |
+          # Run a command to force the outer k3s multicall to precreate the data directory
+          ${{targets.destdir}}/usr/bin/k3s -d "${{targets.subpkgdir}}"/var/lib/rancher/k3s completion bash > /dev/null
+
+  - name: k3s-images
+    description: "k3s images, pre-bundled"
+    dependencies:
+      runtime:
+        - k3s
+    pipeline:
+      - runs: |
+          set -e -u -x
+
+          _arch=$(go env GOARCH)
+
+          images=""
+          while read -r line
+          do
+            repo=$(echo $line | cut -d ':' -f 1)
+            tag=$(echo $line | cut -d ':' -f 2)
+
+            # Replace upstream images with cgr.dev equivalents only if they exist
```

Comment:
This section is mostly kept in as a TODO for now, when the images below have `cgr.dev` variants, we'll uncomment them here. For now, fallback to the upstream variants.

Reply:
ack